File: frontend/legalsaathi/src/app/core/dtos/ChangeRoleToLawyerRequestDTO.ts
import { LawyerSpecialization } from "../models/LawyerSpecialization";

export class ChangeRoleToLawyerRequestDTO {
    lawyerSpecialization: LawyerSpecialization;
    barCouncilId: string;
    experience: number;
    firmName: string;
    officeAddress: string;

    constructor(
        lawyerSpecialization: LawyerSpecialization,
        barCouncilId: string,
        experience: number,
        firmName: string,
        officeAddress: string
    ) {
        this.lawyerSpecialization = lawyerSpecialization;
        this.barCouncilId = barCouncilId;
        this.experience = experience;
        this.firmName = firmName;
        this.officeAddress = officeAddress;
    }
}

File: backend/legalsaathi/src/main/java/com/ascoder1109/legalsaathi/dtos/ChangeRoleToLawyerRequestDTO.java
    package com.ascoder1109.legalsaathi.dtos;

    import com.ascoder1109.legalsaathi.entities.LawyerSpecialization;
    import lombok.Data;


    @Data
    public class ChangeRoleToLawyerRequestDTO {
        private LawyerSpecialization lawyerSpecialization;
        private String barCouncilId;
        private int experience;
        private String firmName;
        private String officeAddress;
    }
